File: leet_easy_c++/leet_206/leet_206b.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* a = NULL;
        ListNode* b = head;
        ListNode* c;
        while(b){
            c = b->next;
            b->next = a;
            a = b;
            b=c;
        }
        return a;
    }
};

File: leet_easy_javascript/leet_206/leet_206b.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

var prev = null;

/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    if(head === null) return null;
    if(head.next === null){
        head.next = prev;
        prev = null;
        return head;
    } 
    var next = head.next;
    head.next = prev;
    prev = head;
    return reverseList(next);
};